feat: allow for some params to not be expanded

FedeAbella opened the pull request:
This allows some extra verbosity at the moment of using `addParams` to `cmd_builder` to be used so that particular parameter value will **not** be expanded when the cmd is built. It's done so as to keep all the other method calls the same and working as intended.

xixiaofinland commented:
The design looks fine to me. But I suggest you adding expanding/no-expanding tests to guard this new feature if not a big hassle. 

Or I can add it afterwards, which is also fine. @FedeAbella

FedeAbella commented:
Absolutely! Don't worry abou it, I'll take care of it. I needed an excuse to learn how tests work in lua (and in this project especifically) anyways :rofl:

FedeAbella commented:
@xixiaofinland could use your help here: I seem to be running into an issue running the tests with `mini.test` where it can't find the treesitter parsers module:
![image](https://github.com/user-attachments/assets/351f49d9-c2ee-4e5f-a3c8-ba8fa4d83a31)

Even though I do have it installed. Any idea what might be going on?

FedeAbella commented:
Ah, I definitely missed that. Thanks!

FedeAbella commented:
Added some tests @xixiaofinland. Let me know if they look enough for you. I've also taken your advice and moved the whole thing to a new `addParamsNoExpand` method.

I think I kind of got the gist of the tests, so I'll see about expanding some other tests later if these look good to you

xixiaofinland commented on tests/test_cmd_builder.lua:
```diff
@@ -72,6 +72,31 @@ T["setup()"]["can supply mutliple params in table in addParams()"] = function()
+  eq(result, expected)
+end
+
+T["setup()"]["can supply flag:value string in more than one addParamsNoExpand, and none will be expanded()"] = function()
```
These tests don't have `addParamsNoExpand()` in test setup but the description says so. Something wrong? @FedeAbella

FedeAbella replied:
Sorry about that, I was multitasking, got one of the tests to work by setting the file, then forgot to apply that to the other ones :( It's fixed now
